File: frontend/src/services/reservationService.js
import api from './api';

export const reservationService = {
    createReservation: (data) => api.post('/reservations', data),

    getMyReservations: () => {
        const user = JSON.parse(localStorage.getItem('user'));
        const userId = user?.id || 1;
        return api.get(`/reservations/user/${userId}`);
    },

    getReservationById: (id) => api.get(`/reservations/${id}`),

    updateReservation: (id, data) => api.put(`/reservations/${id}`, data),

    cancelReservation: (id, reason = 'Cancelled by user') =>
        api.delete(`/reservations/${id}?reason=${encodeURIComponent(reason)}`),

    confirmReservation: (id) => api.put(`/reservations/${id}/confirm`),

    isTableAvailable: (tableId, date) =>
        api.get(`/reservations/${tableId}/available/${date}`),

    getRestaurantReservations: (restaurantId) =>
        api.get(`/reservations/restaurant/${restaurantId}`),

    getReservationsByDate: (restaurantId, date) =>
        api.get(`/reservations/restaurant/${restaurantId}/date/${date}`),
};

File: src/main/java/com/example/demo/service/ReservationService.java
// ПУТЬ: src/main/java/com/example/demo/service/ReservationService.java

package com.example.demo.service;

import com.example.demo.dto.request.ReservationCreateRequestDTO;
import com.example.demo.dto.response.ReservationResponseDTO;
import java.time.LocalDate;
import java.util.List;

public interface ReservationService {
    ReservationResponseDTO createReservation(ReservationCreateRequestDTO request, Long userId);
    ReservationResponseDTO getReservationById(Long id);
    List<ReservationResponseDTO> getUserReservations(Long userId);
    List<ReservationResponseDTO> getRestaurantReservations(Long restaurantId);
    List<ReservationResponseDTO> getReservationsByDate(Long restaurantId, LocalDate date);
    void updateReservation(Long id, ReservationCreateRequestDTO request);
    void cancelReservation(Long id, String reason);
}